change/@ni-nimble-components-df1cc26c-9976-477a-8fa0-f9b72492eaa0.json

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
I did a bunch of manual testing and it was generally very stable and predictable. Nice work!

The one area where the behavior felt inconsistent was with the impact of clicking inside the table to reset the location that keyboard focus should start at. Sometimes I could click in the table and then press Tab and focus would start on the row that I clicked. But other times focus would start on a row that previously had focus.

Here's a video showing some of that inconsistency. This is using the [Grouping story in the latest storybook build](https://60e89457a987cf003efc0a5b-gylhkuqjos.chromatic.com/iframe.html?args=&id=components-table-column-configuration--grouping&viewMode=story). It's hard to see exactly what I'm doing, but sometimes clicking on a row will cause focus to return there the next time I press tab / shift-tab and sometimes it won't. I can't tell if the difference is due to where I click (group vs row), what had focus before, how many times I click, or some other factor.

This is macOS FF but I saw similar issues in Safari.

https://github.com/ni/nimble/assets/10500124/d5fbb84a-8117-4535-bcd6-e72f3af4c86c

In Edge I saw even weirder behavior, where pressing Shift (not even Tab) would cause the table to regain focus. In this video, the first few times the table gets focus, I'm just pressing Shift.

https://github.com/ni/nimble/assets/10500124/25036019-ae89-4d79-9d59-a5def10e165c

Reply:
So far, after testing based on your videos, I've found 2 issues:
- Pressing Tab when a group row is the current table focus results in a JS error. This is probably why you get an unexpected row selected (FF video, 0:18)
- If a cell has previously been keyboard focused via left/right arrows, if you focus another row, then focus the previous row, you may see the green focus ring around the last-focused cell (when actually the entire row should be focused). I think this is what you see in your FF video around 0:42.

Both of those issues happen in Chrome too. So far I haven't been able to figure out why you never got the Ralph Wiggum row selected (around 0:30-0:40) in the Firefox video.

For your Edge video, I'm not sure if there's an issue or not. It depends if you're mouse clicking before pressing Shift or not I guess. Basically, if focus is elsewhere in the browser, then you click the table, the table will be the active/focused element, but not focus-visible (no green ring). At that point, pressing most keyboard keys will trigger the focus-visible green ring (Shift, but also just something like a letter key). I think that's expected behavior given how I'm styling against `row:focus-visible`  / `cell:focus-visible` / etc.

Reply:
Just for visibility, I'll list an issue @mollykreis reported, starting from the Table Anchor Column Storybook:
- Focus the table, arrow down to a row with a link. Right-click a link in another row/cell (brings up the link context menu), then press Esc to close that menu. At this point, the link is active/focused, but not focus-visible (pressing any/most keyboard keys will make it focus visible). However the keyboard nav code thinks the row is focused, so pressing Esc won't leave navigation mode/ refocus the cell like you'd expect.

Not sure if that issue has overlap with the other 3 or not yet.

Reply:
>For your Edge video, I'm not sure if there's an issue or not. It depends if you're mouse clicking before pressing Shift or not I guess.

I was indeed clicking on the table and then pressing Shift, so perhaps it is expected. The part that confused me is that I don't see that behavior in Firefox or Safari. Is it a known difference in focus-visible behavior across browsers? (I wouldn't be surprised if it is, I've seen other differences before).

Reply:
Yeah I don't think there's really a standard for exactly when the focus-visible visuals should be shown, the exact implementation is browser dependent. It's also hard to test them out from autotests (since programmatically focusing elements or sending key events may not trigger those visuals).

So overall I'd say as long as we see the focus visuals during regular keyboard navigation operations, that should be sufficient. That's true for Chrome/Firefox, and presumably you've tested Safari enough to establish it there too.

Reply:
@jattasNI @mollykreis I pushed a [PR update](https://github.com/ni/nimble/pull/2172/commits/e0a3d6e2cd20982809ec54ee531614e1a9aa6d49) that I believe addresses the remaining issues from this thread.

[Storybook build](https://60e89457a987cf003efc0a5b-cckjvqymgt.chromatic.com/?path=/story/components-table--table&args=data:HierarchicalDataSet)

Reply:
Also added some new tests for these cases. Marking as Resolved, feel free to reopen if anyone sees similar/related issues.